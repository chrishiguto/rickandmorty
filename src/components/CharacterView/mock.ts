const characterViewMock = {
  name: 'Rick Sanchez',
  species: 'Human',
  gender: 'Male',
  status: 'Alive',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  about:
    'Rick Sanchez is a male human. He is alive and well. Was last seen at Nov 10, 2017.',
  location: {
    type: 'Planet',
    name: 'Earth (Replacement Dimension)',
    dimension: 'Replacement Dimension',
    created: '2017-11-10T13:08:13.191Z',
    residents: [
      {
        id: '1'
      },
      {
        id: '2'
      }
    ]
  },
  origin: {
    type: 'Planet Two',
    name: 'Second Earth (Same Dimension)',
    dimension: 'Same Dimension',
    residents: [
      {
        id: '1'
      },
      {
        id: '2'
      }
    ]
  },
  isOpen: true
}

export default characterViewMock
