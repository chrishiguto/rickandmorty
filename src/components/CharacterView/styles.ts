import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import * as CharacterCardStyles from 'components/CharacterCard/styles'

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;

  padding: 15rem 1.2rem 1.6rem 1.2rem;

  ${media.greaterThan('small')`
      padding: 15rem 5.6rem 5.6rem 5.6rem;
    `}

  ${media.greaterThan('medium')`
      padding: 2rem 2rem 2rem 5.6rem;
    `}

    ${media.greaterThan('large')`
      max-width: 130rem;
    `}
`

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    height: 100%;

    border-radius: calc(${theme.border.radius} * 2);
    border: 0.2rem solid ${theme.colors.lightBlack};
    box-shadow: 0px 4px 80px 0 #000;

    ${media.greaterThan('medium')`
      flex-direction: row;
    `}
  `}
`

export const Close = styled.button`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    left: 0;
    margin-top: ${theme.spacings.small};
    margin-left: ${theme.spacings.small};

    background: none;
    border: none;
  `}
`

type StripProps = {
  img: string
}

export const Strip = styled.div<StripProps>`
  ${({ theme, img }) => css`
    position: relative;
    display: flex;
    flex-direction: column;

    background-image: url(${img});
    background-size: cover;
    background-position: center center;

    height: 6rem;
    border-top-left-radius: 1.4rem;
    border-top-right-radius: 1.4rem;

    z-index: ${theme.layers.base};

    &::before {
      content: '';
      position: absolute;
      width: 100%;
      height: 100%;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.7);
      backdrop-filter: blur(2.8rem);
      border-top-left-radius: 1.4rem;
      border-top-right-radius: 1.4rem;
      z-index: -1;

      ${media.greaterThan('medium')`
        border-top-left-radius: 1.4rem;
        border-bottom-left-radius: 1.4rem;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    `}
    }

    ${CharacterCardStyles.Wrapper} {
      position: absolute;
      top: -170%;
      left: 50%;
      transform: translateX(-50%);
    }

    ${media.greaterThan('medium')`
      flex: 0.8;
      max-width: 39rem;
      height: 100%;
      border-top-left-radius: 1.4rem;
      border-bottom-left-radius: 1.4rem;
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;

      ${CharacterCardStyles.Wrapper} {
        position: initial;
        transform: none;
        margin: 8.4rem 5.6rem 5.6rem -5.6rem;
        z-index: ${theme.layers.overlay};
      }
    `}
  `}
`

export const StripContent = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: 4rem;
    flex: 1;
    overflow: auto;

    padding: ${theme.spacings.xxlarge} ${theme.spacings.small};
    background-color: rgba(0, 0, 0, 0.98);

    border-bottom-left-radius: 1.4rem;
    border-bottom-right-radius: 1.4rem;

    ${media.greaterThan('medium')`
      gap: 6rem;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
      border-top-right-radius: 1.4rem;
      border-bottom-right-radius: 1.4rem;
      padding: calc(${theme.spacings.xxlarge} + 2rem) calc(${theme.spacings.xxlarge} + 2rem);
    `}
  `}
`

export const Information = styled.div``

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.white};
    line-height: 2.2rem;
    margin-top: ${theme.spacings.xsmall};
  `}
`

export const Type = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.darkGray};
    margin-top: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.small};
    `}
  `}
`

export const Name = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    color: ${theme.colors.white};
    line-height: 1;
    margin: 0.4rem 0 0.6rem 0;

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`

export const Dimension = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.lightGray};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`

export const Residents = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-top: ${theme.spacings.xsmall};
  `}
`

export const Quantity = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    margin-left: ${theme.spacings.xxsmall};
  `}
`
