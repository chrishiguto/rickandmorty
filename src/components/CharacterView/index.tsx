import { format } from 'date-fns'
import { Users, X } from '@styled-icons/feather'

import { CharacterProps } from 'pages'

import Button from 'components/Button'
import CharacterCard from 'components/CharacterCard'
import Heading from 'components/Heading'
import MediaMatch from 'components/MediaMatch'

import {
  getPronoumByGender,
  getStatus,
  getVerbByStatus,
  PronoumProps,
  StatusProps
} from 'utils/about/phrase'

import theme from 'styles/theme'
import * as S from './styles'

export type CharacterViewProps = {
  onClose: () => void
} & CharacterProps

const CharacterView = ({
  name,
  gender,
  image,
  species,
  status,
  location,
  origin,
  onClose
}: CharacterViewProps) => {
  const getAboutPhrase = () => {
    const phraseStatus = status.toLowerCase() as keyof StatusProps
    const phraseGender = gender.toLowerCase() as keyof PronoumProps
    const phraseSpecies = species.toLowerCase()
    const lastSeen = location?.created
      ? format(new Date(location.created), 'MMM dd, yyyy')
      : 'unknown date'

    const firstSentence = `${name} ${getVerbByStatus(
      phraseStatus
    )} ${phraseGender} ${phraseSpecies}`

    const secondSentence = `${
      phraseStatus === 'unknown'
        ? `It can't be told if ${getPronoumByGender(
            phraseGender
          ).toLowerCase()} alive or dead`
        : `${getPronoumByGender(phraseGender)} ${getStatus(phraseStatus)}`
    }`

    const thirdSentence = `Was last seen at ${lastSeen}`

    return `${firstSentence}. ${secondSentence}. ${thirdSentence}.`
  }

  const getPlaceInformation = (
    info: string | undefined | 'unknown',
    fallback: string
  ) => {
    if (!info || info === 'unknown') {
      return fallback
    }

    return info
  }

  return (
    <S.Wrapper>
      <S.Container>
        <MediaMatch lessThan="medium">
          <S.Close onClick={onClose} aria-label="Close character view mobile">
            <X size={32} strokeWidth={2} color="#fff" />
          </S.Close>
        </MediaMatch>
        <S.Strip
          img="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          aria-label={`Background with blurred image of ${name}`}
        >
          <MediaMatch greaterThan="medium">
            <Button
              onClick={onClose}
              style={{
                zIndex: theme.layers.alwaysOnTop,
                margin: theme.spacings.xsmall,
                alignSelf: 'flex-start',
                position: 'absolute'
              }}
              aria-label="Close character view desktop"
            >
              Close
            </Button>
          </MediaMatch>
          <S.StripContent>
            <MediaMatch lessThan="medium">
              <CharacterCard name={name} image={image} species={species} />
            </MediaMatch>
            <MediaMatch
              style={{
                alignItems: 'center',
                justifyContent: 'center',
                height: '100%',
                width: '100%'
              }}
              greaterThan="medium"
            >
              <CharacterCard
                transparentStripe={false}
                size="large"
                name={name}
                image={image}
                species={species}
              />
            </MediaMatch>
          </S.StripContent>
        </S.Strip>
        <S.Content>
          <S.Information>
            <Heading>About</Heading>
            <S.Text>{getAboutPhrase()}</S.Text>
          </S.Information>

          <S.Information>
            <Heading>Origin</Heading>
            <S.Type>
              {getPlaceInformation(origin?.type, 'Unknown Planet')}
            </S.Type>
            <S.Name>{getPlaceInformation(origin?.name, 'Unknown')}</S.Name>
            <S.Dimension>
              {getPlaceInformation(origin?.dimension, 'Unknown Dimension')}
            </S.Dimension>
            {!!origin?.residents && (
              <S.Residents>
                <Users size={14} color="#8c8c8c" strokeWidth={2} />
                <S.Quantity>{origin.residents.length} residents</S.Quantity>
              </S.Residents>
            )}
          </S.Information>

          <S.Information>
            <Heading>Location</Heading>
            <S.Type>
              {getPlaceInformation(location?.type, 'Unknown Planet')}
            </S.Type>
            <S.Name>{getPlaceInformation(location?.name, 'Unknown')}</S.Name>
            <S.Dimension>
              {getPlaceInformation(location?.dimension, 'Unknown Dimension')}
            </S.Dimension>
            {!!location?.residents && (
              <S.Residents>
                <Users size={14} color="#8c8c8c" strokeWidth={2} />
                <S.Quantity>{location.residents.length} residents</S.Quantity>
              </S.Residents>
            )}
          </S.Information>
        </S.Content>
      </S.Container>
    </S.Wrapper>
  )
}

export default CharacterView
