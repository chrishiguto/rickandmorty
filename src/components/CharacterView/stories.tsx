import { Story, Meta } from '@storybook/react/types-6-0'

import CharacterView, { CharacterViewProps } from '.'
import characterViewMock from './mock'

export default {
  title: 'CharacterView',
  component: CharacterView,
  args: {
    ...characterViewMock
  }
} as Meta

export const Mobile: Story<CharacterViewProps> = (args) => (
  <div style={{ paddingTop: 14 }}>
    <CharacterView {...args} />
  </div>
)

Mobile.parameters = {
  viewport: {
    defaultViewport: 'mobile1'
  }
}

export const Desktop: Story<CharacterViewProps> = (args) => (
  <div style={{ paddingLeft: 100, height: '80vh' }}>
    <CharacterView {...args} />
  </div>
)
