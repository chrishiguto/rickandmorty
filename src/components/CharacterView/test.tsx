import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CharacterView from '.'
import characterViewMock from './mock'

jest.mock('components/CharacterCard', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Character Card">{children}</div>
  }
}))

describe('<CharacterView />', () => {
  it('should render the character view', () => {
    const { container } = renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Alive"
        gender="Male"
      />
    )

    expect(screen.getAllByTestId(/mock character card/i)).toHaveLength(2)
    expect(
      screen.getByLabelText(/background with blurred image of rick sanchez/i)
    ).toHaveStyleRule('background-image', `url(${characterViewMock.image})`)
    expect(screen.getByRole('heading', { name: /about/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /origin/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /location/i })
    ).toBeInTheDocument()

    expect(screen.getByText(characterViewMock.about)).toBeInTheDocument()

    expect(screen.getByText(characterViewMock.origin.type)).toBeInTheDocument()
    expect(screen.getByText(characterViewMock.origin.name)).toBeInTheDocument()
    expect(
      screen.getByText(characterViewMock.origin.dimension)
    ).toBeInTheDocument()

    expect(
      screen.getByText(characterViewMock.location.type)
    ).toBeInTheDocument()
    expect(
      screen.getByText(characterViewMock.location.name)
    ).toBeInTheDocument()
    expect(
      screen.getByText(characterViewMock.location.dimension)
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render unknown type/dimension/planet if there is not results', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Alive"
        gender="Male"
        location={null}
        origin={null}
      />
    )

    expect(screen.getAllByText(/unknown planet/i)).toHaveLength(2)
    expect(screen.getAllByText(/unknown dimension/i)).toHaveLength(2)
    expect(screen.getAllByText(/unknown/i)).toHaveLength(7)
  })

  it('should render a female about text', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Alive"
        gender="Female"
      />
    )

    expect(screen.getByText(/female/i)).toBeInTheDocument()
    expect(screen.getByText(/she is/i)).toBeInTheDocument()
  })

  it('should render a genderless about text', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Alive"
        gender="Genderless"
      />
    )

    expect(screen.getByText(/genderless/i)).toBeInTheDocument()
    expect(screen.getByText(/they are/i)).toBeInTheDocument()
  })

  it('should render a unknown gender about text', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Alive"
        gender="unknown"
      />
    )

    expect(screen.getByText(/unknown/i)).toBeInTheDocument()
    expect(screen.getByText(/they are/i)).toBeInTheDocument()
  })

  it('should render a dead verb about text', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="Dead"
        gender="Male"
      />
    )

    expect(screen.getByText(/dead/i)).toBeInTheDocument()
    expect(screen.getByText(/was a/i)).toBeInTheDocument()
  })

  it('should render a unknown verb about text', () => {
    renderWithTheme(
      <CharacterView
        {...characterViewMock}
        onClose={jest.fn}
        status="unknown"
        gender="Male"
      />
    )

    expect(screen.getByText(/is a/i)).toBeInTheDocument()
    expect(
      screen.getByText(/it can't be told if he is alive or dead/i)
    ).toBeInTheDocument()
  })
})
