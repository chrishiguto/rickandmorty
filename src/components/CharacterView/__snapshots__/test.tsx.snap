// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterView /> should render the character view 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  background: transparent;
  padding: 0.8rem calc(1.6rem - 0.2rem);
  font-size: 1.6rem;
  font-weight: 700;
  border: 2px solid #ffffff;
  border-radius: 0.8rem;
  color: #ffffff;
  outline: none;
  -webkit-transition: box-shadow,background-color 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: box-shadow,background-color 0.3s cubic-bezier(0.4,0,0.2,1);
}

.c7:hover {
  background-color: #8C8C8C35;
}

.c7:focus-visible {
  box-shadow: 0 0 0 0.3rem #cbd736;
}

.c10 {
  color: #cbd736;
  font-weight: 400;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.8rem;
  -moz-letter-spacing: 0.8rem;
  -ms-letter-spacing: 0.8rem;
  letter-spacing: 0.8rem;
  font-size: 1.4rem;
}

.c2 {
  display: none;
}

.c6 {
  display: none;
}

.c0 {
  width: 100%;
  height: 100%;
  padding: 15rem 1.2rem 1.6rem 1.2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  border-radius: calc(0.8rem * 2);
  border: 0.2rem solid #1e1e1e;
  box-shadow: 0px 4px 80px 0 #000;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  margin-top: 2.4rem;
  margin-left: 2.4rem;
  background: none;
  border: none;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: url(https://rickandmortyapi.com/api/character/avatar/1.jpeg);
  background-size: cover;
  background-position: center center;
  height: 6rem;
  border-top-left-radius: 1.4rem;
  border-top-right-radius: 1.4rem;
  z-index: 10;
}

.c5::before {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0,0,0,0.7);
  -webkit-backdrop-filter: blur(2.8rem);
  backdrop-filter: blur(2.8rem);
  border-top-left-radius: 1.4rem;
  border-top-right-radius: 1.4rem;
  z-index: -1;
}

.c5 .sc-hKFxyN {
  position: absolute;
  top: -170%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  padding: 5.6rem 2.4rem;
  background-color: rgba(0,0,0,0.98);
  border-bottom-left-radius: 1.4rem;
  border-bottom-right-radius: 1.4rem;
}

.c11 {
  font-size: 1.6rem;
  color: #ffffff;
  line-height: 2.2rem;
  margin-top: 1.6rem;
}

.c12 {
  font-size: 1.6rem;
  color: #8C8C8C;
  margin-top: 1.6rem;
}

.c13 {
  font-size: 2.0rem;
  color: #ffffff;
  line-height: 1;
  margin: 0.4rem 0 0.6rem 0;
}

.c14 {
  font-size: 1.6rem;
  color: #D3D3D3;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1.6rem;
}

.c16 {
  font-size: 1.4rem;
  color: #8C8C8C;
  margin-left: 0.8rem;
}

@media (max-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:450px) {
  .c0 {
    padding: 15rem 5.6rem 5.6rem 5.6rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2rem 2rem 2rem 5.6rem;
  }
}

@media (min-width:1170px) {
  .c0 {
    max-width: 130rem;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c5::before {
    border-top-left-radius: 1.4rem;
    border-bottom-left-radius: 1.4rem;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex: 0.8;
    -ms-flex: 0.8;
    flex: 0.8;
    max-width: 39rem;
    height: 100%;
    border-top-left-radius: 1.4rem;
    border-bottom-left-radius: 1.4rem;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }

  .c5 .sc-hKFxyN {
    position: initial;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    margin: 8.4rem 5.6rem 5.6rem -5.6rem;
    z-index: 30;
  }
}

@media (min-width:768px) {
  .c9 {
    gap: 6rem;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    border-top-right-radius: 1.4rem;
    border-bottom-right-radius: 1.4rem;
    padding: calc(5.6rem + 2rem) calc(5.6rem + 2rem);
  }
}

@media (min-width:768px) {
  .c12 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .c13 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c14 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <button
        aria-label="Close character view mobile"
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="c4"
          color="#fff"
          fill="none"
          focusable="false"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
    </div>
    <div
      aria-label="Background with blurred image of Rick Sanchez"
      class="c5"
    >
      <div
        class="c6"
      >
        <button
          aria-label="Close character view desktop"
          class="c7"
          style="z-index: 50; margin: 1.6rem; align-self: flex-start; position: absolute;"
        >
          Close
        </button>
      </div>
      <div
        class="c8"
      >
        <div
          class="c2"
        >
          <div
            data-testid="Mock Character Card"
          />
        </div>
        <div
          class="c6"
          style="align-items: center; justify-content: center; height: 100%; width: 100%;"
        >
          <div
            data-testid="Mock Character Card"
          />
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class=""
      >
        <h2
          class="c10"
        >
          About
        </h2>
        <p
          class="c11"
        >
          Rick Sanchez is a male human. He is alive and well. Was last seen at Nov 10, 2017.
        </p>
      </div>
      <div
        class=""
      >
        <h2
          class="c10"
        >
          Origin
        </h2>
        <p
          class="c12"
        >
          Planet Two
        </p>
        <p
          class="c13"
        >
          Second Earth (Same Dimension)
        </p>
        <p
          class="c14"
        >
          Same Dimension
        </p>
        <div
          class="c15"
        >
          <svg
            aria-hidden="true"
            class="c4"
            color="#8c8c8c"
            fill="none"
            focusable="false"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 21v-2a4 4 0 00-4-4H5a4 4 0 00-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M23 21v-2a4 4 0 00-3-3.87M16 3.13a4 4 0 010 7.75"
            />
          </svg>
          <p
            class="c16"
          >
            2
             residents
          </p>
        </div>
      </div>
      <div
        class=""
      >
        <h2
          class="c10"
        >
          Location
        </h2>
        <p
          class="c12"
        >
          Planet
        </p>
        <p
          class="c13"
        >
          Earth (Replacement Dimension)
        </p>
        <p
          class="c14"
        >
          Replacement Dimension
        </p>
        <div
          class="c15"
        >
          <svg
            aria-hidden="true"
            class="c4"
            color="#8c8c8c"
            fill="none"
            focusable="false"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 21v-2a4 4 0 00-4-4H5a4 4 0 00-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M23 21v-2a4 4 0 00-3-3.87M16 3.13a4 4 0 010 7.75"
            />
          </svg>
          <p
            class="c16"
          >
            2
             residents
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`
